### PR TITLE
add ccm backward capability notice and latest banner

### PR DESCRIPTION
**Pull request by guangbochen**
**Description:**
Add ccm backward capability notice and the latest banner

**Preview:**
https://64fef026130f945cc6f25bd1--harvester-preview.netlify.app/v1.2/rancher/cloud-provider#backward-compatibility-notice

### CODE REVIEW

File: docs/rancher/cloud-provider.md [hunk 1]
```diff
@@ -24,6 +24,16 @@ In this page we will learn:
 - How to deploy the Harvester cloud provider in both RKE1 and RKE2 cluster.
 - How to use the [Harvester load balancer](#load-balancer-support).
 
+### Backward Compatibility Notice
+
+:::note
+
+If you're using the Harvester cloud provider version **v0.2.2** or higher, please note a known backward compatibility issue. If your Harvester version is below **v1.2.0** and you intend to use newer RKE2 versions (i.e., >= `v1.26.6+rke2r1`, `v1.25.11+rke2r1`, `v1.24.15+rke2r1`), it is essential to upgrade your Harvester cluster to v1.2.0 or a higher version before proceeding with the upgrade of the guest Kubernetes cluster or Harvester cloud provider.
```
Comment (LucasSaintarbor) [inline docs/rancher/cloud-provider.md:31]:
```suggestion
Please note a known backward compatibility issue if you're using the Harvester cloud provider version **v0.2.2** or higher. If your Harvester version is below **v1.2.0** and you intend to use newer RKE2 versions (i.e., >= `v1.26.6+rke2r1`, `v1.25.11+rke2r1`, `v1.24.15+rke2r1`), it is essential to upgrade your Harvester cluster to v1.2.0 or a higher version before proceeding with the upgrade of the guest Kubernetes cluster or Harvester cloud provider.
```

File: docs/rancher/node/rke2-cluster.md [hunk 2]
```diff
@@ -21,6 +21,17 @@ You can now provision RKE2 Kubernetes clusters on top of the Harvester cluster i
 
 :::
 
+### Backward Compatibility Notice
+
+:::note
+
+If you're using the Harvester cloud provider version **v0.2.2** or higher, please note a known backward compatibility issue. If your Harvester version is below **v1.2.0** and you intend to use newer RKE2 versions (i.e., >= `v1.26.6+rke2r1`, `v1.25.11+rke2r1`, `v1.24.15+rke2r1`), it is essential to upgrade your Harvester cluster to v1.2.0 or a higher version before proceeding with the upgrade of the guest Kubernetes cluster or Harvester cloud provider.
```
Comment (LucasSaintarbor) [inline docs/rancher/node/rke2-cluster.md:28]:
```suggestion
Please note a known backward compatibility issue if you're using the Harvester cloud provider version **v0.2.2** or higher.  If your Harvester version is below **v1.2.0** and you intend to use newer RKE2 versions (i.e., >= `v1.26.6+rke2r1`, `v1.25.11+rke2r1`, `v1.24.15+rke2r1`), it is essential to upgrade your Harvester cluster to v1.2.0 or a higher version before proceeding with the upgrade of the guest Kubernetes cluster or Harvester cloud provider.
```

**Review by LucasSaintarbor — APPROVED** [review]
Small NIT. LGTM, thanks!

**Review by yasker — APPROVED** [review]

**Review by bk201 — APPROVED** [review]
LGTM, thanks!

**Pull request merged**